tests/Microsoft.Identity.Test.Unit.net45/PublicApiTests/ClientCredentialWithCertTest.cs

```diff
@@ -139,6 +159,49 @@ public async Task JsonWebTokenWithX509PublicCertSendCertificateOnBehalfOfTestAsy
             }
         }
 
+        [TestMethod]
+        [Description("Test for client assertion with X509 public certificate using sendCertificate")]
+        public async Task JsonWebTokenWithX509PublicCertSendCertificateSilentTestAsync()
+        {
+            using (var harness = CreateTestHarness())
+            {
+                SetupMocks(harness.HttpManager, "https://login.microsoftonline.com/my-utid/");
+                var certificate = new X509Certificate2(
+                    ResourceHelper.GetTestResourceRelativePath("valid_cert.pfx"),
+                    MsalTestConstants.DefaultPassword);
+
+                var app = ConfidentialClientApplicationBuilder.Create(MsalTestConstants.ClientId)
+                                                              .WithAuthority(
+                                                                  new System.Uri("https://login.microsoftonline.com/my-utid"),
+                                                                  true).WithRedirectUri(MsalTestConstants.RedirectUri)
+                                                              .WithHttpManager(harness.HttpManager)
+                                                              .WithCertificate(certificate).BuildConcrete();
+
+
+
+                _tokenCacheHelper.PopulateCache(app.UserTokenCacheInternal.Accessor);
+                app.UserTokenCacheInternal.Accessor.DeleteAccessToken(
```

Comment:
There is a method that does GetAllAccessTokens - use it and make sure a single token is returned. Then delete that token. Test will look cleaner.